src/GUniLPOracle.t.sol

```diff
@@ -586,4 +592,20 @@ contract GUniLPOracleTest is DSTest {
         assertLt(sqrtPriceX96, 1 << 160);
     }
 
+    function test_sqrt_price_ratio_fuzz(uint256 p0, uint256 dec0, uint256 p1, uint256 dec1) public {
+        p0 %= MAX_PRICE;
+        if (p0 < WAD) p0 = MIN_PRICE;
+        p1 %= MAX_PRICE;
+        if (p1 < WAD) p1 = MIN_PRICE;
+        dec0 %= MAX_DEC;
+        dec1 %= MAX_DEC;
+
+        uint256 UNIT_0 = 10 ** dec0;
+        uint256 UNIT_1 = 10 ** dec1;
+
+        uint160 sqrtPriceX96 = uint160(sqrt2(mul(mul(p0, UNIT_1), (1 << 96)) / (mul(p1, UNIT_0))) << 48);
```

Comment:
This should probably be safe cast to uint160 as in the actual oracle code.

Reply:
Done in 965dabe.